File: javascript/day1.js
#!/usr/local/bin/node

'use strict';

var fs = require('fs');

let instructions = fs.readFileSync('../day1_input.txt', 'utf-8').split(''),
    instruction_values = {'(': 1, ')': -1};

var final_floor = instructions
    .map(function (char) {
        return instruction_values[char];
    })
    .reduce(function (prev, current) {
        return prev + current;
    });

console.log('Final floor:', final_floor);


// Part two

var floor = 0,
    basement_instruction = 0;

for (let char of instructions) {
    floor += instruction_values[char];
    basement_instruction += 1;
    if (floor === -1) {
        console.log('First basement instruction:', basement_instruction);
        break;
    }
}

File: day1.py
#!/usr/local/bin/python3

with open('day1_input.txt') as f:
    instructions = f.read().rstrip()

floor = 0

for char in instructions:
    if char == '(':
        floor += 1
    else:
        floor -= 1

print('final floor:', floor)


# Part two


floor = 0

for idx, char in enumerate(instructions, start=1):
    if char == '(':
        floor += 1
    else:
        floor -= 1
    if floor == -1:
        print('first basement instruction:', idx)
        break
